lib/propolis/src/lifecycle.rs

```diff
@@ -51,13 +51,21 @@ pub trait Lifecycle: Send + Sync + 'static {
     /// The device is not required to finish pausing inline. Instead, its
     /// implementation of [`Lifecycle::paused`] should return a future that
     /// completes only when the device is paused.
+    ///
+    /// WARNING: This function may only be called after pausing all of a VM's
```

Comment:
Thanks for adding the docs here.  The context is definitely nice for implementers.